src/routes/DigitalAssets/routes/EditAsset/sagas/editAsset.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+
+import { put, select, takeEvery } from 'redux-saga/effects'
+import { backOrFallback } from 'routes/modules'
+
+import {
+  selectEditAsset,
+  selectDigitalAsset,
+} from 'store/stateSelectors'
+
+import {
+  setField,
+  setFieldError,
+  clearFieldError,
+  OPEN_VIEW,
+  SUBMIT_ASSET_FORM,
+  typeof openView as openViewType,
+} from '../modules/editAsset'
+
+import {
+  updateAsset,
+} from '../../../modules/digitalAssets'
+
+function* clearFieldsError(): Saga<void> {
+  yield put(clearFieldError('name'))
+  yield put(clearFieldError('symbol'))
+  yield put(clearFieldError('decimals'))
+}
+
+function* onAssetFormSumbit(): Saga<void> {
+  const {
+    formFields: {
+      address,
+      name,
+      symbol,
+      decimals,
+    },
+  }: ExtractReturn<typeof selectEditAsset> = yield select(selectEditAsset)
+
+  const contractName = name.trim()
+  const contractSymbol = symbol.trim()
+  const contractDecimals = parseInt(decimals, 10)
+
+  yield* clearFieldsError()
+
+  // Check, do we have this asset
+  const foundAsset: ?DigitalAsset = yield select(selectDigitalAsset, address)
+  if (!(foundAsset && foundAsset.isCustom)) {
+    yield put(setFieldError('address', 'Asset not found'))
+    return
+  }
+
+  if (contractName.length === 0) {
+    yield put(setFieldError('name', 'Valid digital asset name is required'))
+  }
+
+  if (contractSymbol.length === 0 || contractSymbol.length > 10) {
+    yield put(setFieldError('symbol', 'Valid digital asset symbol is required'))
+  }
+
+  if (Number.isNaN(contractDecimals) ||
+    contractDecimals <= 0 ||
+    contractDecimals > 127) {
+    yield put(
+      setFieldError('decimals', 'Digital asset decimals should be a number between 1...127')
+    )
+  }
+
+  const {
+    invalidFields: {
+      address: addressError,
+      name: nameError,
+      symbol: symbolError,
+      decimals: decimalsError,
+    },
+  }: ExtractReturn<typeof selectEditAsset> = yield select(selectEditAsset)
```

Comment:
`yield select(selectEditAsset)` is not covered by flow (it can't cover such generators)
Here we need to fix it, by adding force `ExtractReturn<typeof selectEditAsset>`
After this types of all fields address, symbol... are calculated automatically (as string)